src/components/auth/utils/currencies/CurrenciesItems.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { MenuItem, TextField, styled } from '@mui/material';
+
+const CurrencyField = styled(TextField)((props) => ({
+  width: '100%',
+  backgroundColor: props.theme.colors.background.primary,
+  borderRadius: props.theme.borderRadius,
+  '& label': {
+    color: props.theme.colors.text.darker,
+  },
+  '& label.Mui-focused': {
+    color: props.theme.colors.main.violet,
+  },
+  '& .MuiOutlinedInput-root': {
+    '& fieldset': {
+      borderRadius: props.theme.borderRadius,
+      border: `1px solid ${props.theme.colors.border.item}`,
+    },
+    '&:hover fieldset': {
+      border: `1px solid ${props.theme.colors.text.darker}`,
+    },
+    '&.Mui-focused fieldset': {
+      border: `1px solid ${props.theme.colors.main.violet}`,
+    },
+    '& .MuiInputBase-input': {
+      color: props.theme.colors.text.primary,
+      borderRadius: props.theme.borderRadius,
+      backgroundColor: props.theme.colors.background.primary,
+    },
+    '& .MuiSelect-icon': {
+      fill: props.theme.colors.main.violet,
+    },
+  },
+}));
+
+function CurrenciesItems({ names, currency, setCurrency }) {
+  let results = [];
```

Comment:
It would be more typical to use `results = names.map` here instead of a for loop. Also, you don't need to use `let` for an array to `push` into it, so you would use `const results` instead.